Zend/Optimizer/zend_ssa.h

```diff
@@ -122,16 +122,16 @@ typedef struct _zend_ssa_var {
 
 typedef struct _zend_ssa_var_info {
 	uint32_t               type; /* inferred type (see zend_inference.h) */
+	unsigned char          has_range : 1;
+	unsigned char          is_instanceof : 1; /* 0 - class == "ce", 1 - may be child of "ce" */
+	unsigned char          recursive : 1;
+	unsigned char          use_as_double : 1;
+	unsigned char          delayed_fetch_this : 1;
+	unsigned char          avoid_refcounting : 1;
+	unsigned char          guarded_reference : 1;
+	unsigned char          indirect_reference : 1; /* IS_INDIRECT returned by FETCH_DIM_W/FETCH_OBJ_W */
```

Comment:
Showing my lack of C knowledge. But I suppose all of those fields are basically boolean fields because they only use 1 bit? However, if we would use ``bool``, due to it being typedef to ``int`` we would be using a whole byte for each field?

Reply:
@Girgias Using `bool` indeed uses an entire byte. Compilers do this because otherwise there needs to be an additional bitwise-and when space is usually not the limiting factor.

> due to it being typedef to int

Not that it matters in this context, but `bool` (typealias of `_Bool`) is it's own thing, guaranteeing coercion to `0`/`1`. 

https://godbolt.org/z/jf1eKEcen

```c
char x = getChar();
// call    getChar()
// mov     BYTE PTR [rbp-1], al

bool y = getChar();
// call    getChar()
// test    al, al
// setne   al
// mov     BYTE PTR [rbp-2], al
```

Sorry if you already knew this :wink: 

@nielsdos We should however probably use `bool x:1` here as semantics for unsigned bitfields are still wrapping, which is not what we want here.

https://godbolt.org/z/fz3ezGEoK

Reply:
@iluuu1994 Then it probably was already *potentially* wrong (*) when it used `unsigned int`? I'll change it to bool for an additional reason too: I just looked up the C99 standard: only signed/unsigned int and bool are allowed as bitfield types anyway, for other types it's implementation-defined. I checked on MSVC, ICC, Clang, GCC and bool are all 1 byte long so we keep the space-saving by switching to bool.

(*) Although it seems like the values that are assigned to those fields won't cause any problems, but it's probably best to take a defensive approach here. Especially considering the C99 standard.

Reply:
Sounds good, thanks for having a look!